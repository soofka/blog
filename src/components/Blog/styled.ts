import styledComponents from 'styled-components';

export const StyledBlog = styledComponents.div`
  width: 100%;
  height: 100%;
  font-size: ${props => props.theme.baseFontSize};
  line-height: ${props => props.theme.baseLineHeight};
`;

export default StyledBlog;
