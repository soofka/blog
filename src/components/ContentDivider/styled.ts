import styledComponents from 'styled-components';

export const StyledContentDivider = styledComponents.hr`
  border-color: ${props => props.theme.lighterBlue} !important;
`;

export default StyledContentDivider;
