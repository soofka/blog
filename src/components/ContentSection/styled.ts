import styledComponents from 'styled-components';

export const StyledContentSection = styledComponents.div`
  float: left;
  margin: 0 0 0 ${props => props.theme.infoSectionWidth};
  padding: ${props => props.theme.paddingM};
  width: 100%;
  height: auto;
  min-height: 100%;
  box-sizing: border-box;
  color: ${props => props.theme.lighterBlack};
  background-color: ${props => props.theme.white};
  box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.white}`};

  a {
    color: ${props => props.theme.lighterBlack};

    &:hover,
    &:active {
      color: ${props => props.theme.black};
    }
  }

@media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
  margin: 0;
  padding: ${props => props.theme.paddingS};
  width: 100%;
}`;

export default StyledContentSection;
