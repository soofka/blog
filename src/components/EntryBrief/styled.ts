import styledComponents from 'styled-components';

export const StyledEntryBrief = styledComponents.div`
  float: left;
  width: 100%;
  height: ${props => 24 * props.theme.baseDistance}px;
  margin: ${props => props.theme.marginS} 0 ${props => props.theme.marginM} 0;
  padding: 0;
  border: 2px solid ${props => props.theme.white};
  outline: 1px solid ${props => props.theme.lighterBlack};
  background-repeat: no-repeat;
  background-position: center center;
  background-attachment: scroll;
  background-size: cover;
`;

export default StyledEntryBrief;
