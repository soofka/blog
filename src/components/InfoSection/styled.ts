import styledComponents from 'styled-components';

export const StyledInfoSection = styledComponents.div`
  position: fixed;
  top: 0;
  bottom: 0;
  box-sizing: border-box;
  width: ${props => props.theme.infoSectionWidth};
  height: auto;
  min-height: 100%;
  padding: ${props => props.theme.paddingS};
  color: ${props => props.theme.darkerWhite};
  background-color: ${props => props.theme.blue};
  box-shadow: ${props => `0 0 ${props.theme.baseDistance}px 0 ${props.theme.darkerBlue}`};

  a {
    color: ${props => props.theme.darkerWhite};
    text-decoration: none;

    &:hover,
    &:active {
      color: ${props => props.theme.white};
    }
  }

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    position: static;
    display: inline-block;
    width: 100%;
    min-height: 0;
}
`;

export default StyledInfoSection;

