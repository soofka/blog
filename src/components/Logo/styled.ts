import styledComponents from 'styled-components';

export const StyledLogo = styledComponents.img`
  display: block;
  margin: ${props => props.theme.marginL} auto;
  padding: 0;
  width: ${props => 28 * props.theme.baseDistance}px;
  height: ${props => 28 * props.theme.baseDistance}px;
  border-radius: ${props => 4 * props.theme.baseDistance}px;

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    float: left;
    margin: 0;
    padding: 0;
    width: ${props => 8 * props.theme.baseDistance}px;
    height: ${props => 8 * props.theme.baseDistance}px;
    border-radius: ${props => props.theme.baseDistance}px;
  }
`;

export default StyledLogo;
