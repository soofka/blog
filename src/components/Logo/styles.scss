@import 'components/Blog/variables.scss';

$logoMobileSize: 8 * $baseDistance;
$logoDesktopSize: 28 * $baseDistance;

.logo {
  display: block;
  margin: $marginL auto;
  padding: 0;
  width: $logoDesktopSize;
  height: $logoDesktopSize;
  border-radius: $logoDesktopSize / 8;
}

@media (max-width: $mobileBreakPointWidth) {
  .logo {
    float: left;
    margin: 0;
    padding: 0;
    width: $logoMobileSize;
    height: $logoMobileSize;
    border-radius: $logoMobileSize / 8;
  }
}
