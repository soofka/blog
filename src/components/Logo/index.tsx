import * as React from 'react';
import { Link } from 'react-router-dom';

import AssetsProvider from 'common/AssetsProvider';
import RoutingProvider from 'common/RoutingProvider';

import './styles.scss';

export const Logo = (): JSX.Element => {
  return (
    <Link to={RoutingProvider.getHomeNiceUrlBasePath()}>
      <img src={`${AssetsProvider.getImagesBasePath()}/soofka-logo.jpg`} className="logo" />
    </Link>
  );
};

export default Logo;
