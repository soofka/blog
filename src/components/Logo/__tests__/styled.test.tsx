import * as React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import theme from 'common/theme';
import { StyledLogo } from 'components/Logo/styled';

describe('<StyledLogo/>', () => {

  it('is styled correctly', () => {
    expect(shallow(<StyledLogo theme={theme} />)).toMatchSnapshot();
  });

});
