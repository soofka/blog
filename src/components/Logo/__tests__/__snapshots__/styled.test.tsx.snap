// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledLogo/> is styled correctly 1`] = `
.c0 {
  display: block;
  margin: 32px auto;
  padding: 0;
  width: 224px;
  height: 224px;
  border-radius: 32px;
}

@media (max-width:968px) {
  .c0 {
    float: left;
    margin: 0;
    padding: 0;
    width: 64px;
    height: 64px;
    border-radius: 8px;
  }
}

<img
  className="c0"
/>
`;
